library/std/src/os/freebsd/fs.rs

```diff
@@ -74,6 +75,10 @@ pub trait MetadataExt {
 impl MetadataExt for Metadata {
     #[allow(deprecated)]
     fn as_raw_stat(&self) -> &raw::stat {
+        // The methods below use libc::stat, so they work fine when libc is built with FreeBSD 12 ABI.
+        // This method would just return nonsense.
+        #[cfg(freebsd12)]
+        panic!("as_raw_stat not supported with FreeBSD 12 ABI");
         unsafe { &*(self.as_inner().as_inner() as *const libc::stat as *const raw::stat) }
```

Comment:
you also need to make this code `#[cfg(not(freebsd12))]` otherwise you'll get warnings or even errors when compiling libstd